uxarray/grid.py

```diff
@@ -528,6 +531,33 @@ def _build_edge_node_connectivity(self):
         setattr(self, "Mesh2_edge_nodes", self.ds['Mesh2_edge_nodes'])
         setattr(self, "nMesh2_edge", edge_nodes_unique.shape[0])
 
+    def build_face_edges_connectivity(self):
```

Comment:
My thought is: it's good to make it a user API so that user can build the connectivity under their need. Especially if our MPAS will only provide the `edge_nodes_connectivity` but not the `face_edge_connectivity`

Reply:
See my comments below, we should be calling the `build_face_edge_connectivity` by default. Nothing is preventing the user from using this function on their own accord from the internal api, however it's not suggested / practical for them to do so.

By default, our `Grid` constructor should build all connectivity variables that we support. We can add an argument (something like `build_connectivity`) which can indicate whether the user wants to skip the construction of edge nodes, face edges, etc.

Reply:
We had that discussion last summer, the conclusion is only called the `Grid.build_face_edge_connectivity` if it's needed. However, we can make it an internal API if that's more appropriate.

And I wonder why we want to ban users from calling this connectivity. I thought it was the functionality they want. And we already has the check inside the function to prevent duplicate call,

Reply:
I don't recall that discussion, but that was a while ago. Do you remember if it was discussed in an issue or in our meeting notes? 

I'm open to finding a good solution. As I described below, I think having an argument in the grid object to determine whether we want to build the connectivity could be useful. However I think we should give this design some more thought, since I also want to restructure and move these functions into a separate file that can then be called by the `Grid` class. Then, we can consider both an internal (modifies `Grid` dataset) and user (returns the desired connectivity) versions of the function.

If you're not opposed, let's still to keeping this as an internal for now until we set up the `connectivity.py` file